### PR TITLE
feat: impl `Environment` for `RemoteEnvironment`

### PR DESCRIPTION
**Pull request by ysndr**
* implemented by delegating all methods to an inner `ManagedEnvironment` instance
* resolving `--remote` flag
* implement dispaly conversions for `RemoteEnvironment`s where still outstanding
* add tests for core environment operations

### CODE REVIEW

File: cli/flox-rust-sdk/src/models/environment/remote_environment.rs [hunk 1]
```diff
@@ -1,22 +1,76 @@
...
+            owner: env_ref.owner().clone(),
+            name: env_ref.name().clone(),
+            version: Version::<1>,
+        };
+        let mut inner = ManagedEnvironment::open(flox, pointer, path)
+            .map_err(|e| RemoteEnvironmentError::OpenManagedEnvironment(Box::new(e)))?;
+
+        inner
+            .pull(true)
+            .map_err(RemoteEnvironmentError::GetLatestVersion)?;
```
Comment (mkenigs) [inline cli/flox-rust-sdk/src/models/environment/remote_environment.rs:48]: nonblocking: shouldn't `ManagedEnvironment::open` with no lock already perform a pull?
```suggestion
```

File: cli/flox-rust-sdk/src/models/environment/remote_environment.rs [hunk 2]
```diff
@@ -26,7 +80,11 @@ impl Environment for RemoteEnvironment {
         packages: Vec<String>,
         flox: &Flox,
     ) -> Result<InstallationAttempt, EnvironmentError2> {
-        todo!()
+        let result = self.inner.install(packages, flox).await?;
+        self.inner
+            .push(false)
+            .map_err(RemoteEnvironmentError::UpdateUpstream)?;
+        Ok(result)
```
Comment (mkenigs) [inline cli/flox-rust-sdk/src/models/environment/remote_environment.rs:87]: blocking (unless you want to followup): are you symlinking somewhere?

File: cli/flox-rust-sdk/src/models/environment/remote_environment.rs [hunk 3]
```diff
@@ -26,7 +80,11 @@ impl Environment for RemoteEnvironment {
         packages: Vec<String>,
         flox: &Flox,
     ) -> Result<InstallationAttempt, EnvironmentError2> {
-        todo!()
+        let result = self.inner.install(packages, flox).await?;
+        self.inner
+            .push(false)
+            .map_err(RemoteEnvironmentError::UpdateUpstream)?;
```
Comment (mkenigs) [inline cli/flox-rust-sdk/src/models/environment/remote_environment.rs:86]: nonblocking: we could consider deleting the git branch for the temporary environment

**Review by mkenigs — CHANGES_REQUESTED** [review]

File: cli/flox-rust-sdk/src/models/environment/remote_environment.rs (hunk 2 above)
Comment (ysndr) [reply cli/flox-rust-sdk/src/models/environment/remote_environment.rs:87]: the `ManagedEnv` will set a out link.. so activate should work already

File: cli/flox-rust-sdk/src/models/environment/remote_environment.rs (hunk 3 above)
Comment (ysndr) [reply cli/flox-rust-sdk/src/models/environment/remote_environment.rs:86]: yeah we create some stale braches, I'd leave that for a separeate issue

File: cli/flox-rust-sdk/src/models/environment/remote_environment.rs (hunk 2 above)
Comment (mkenigs) [reply cli/flox-rust-sdk/src/models/environment/remote_environment.rs:87]: I haven't checked, but I'm expecting remote environments should create a `<system>.<name>` link, and managed should create `<system>.<name>.encode(<path of user dir>)`. If you're relying on `ManagedEnvironment`, is it giving it the wrong name?

File: cli/flox-rust-sdk/src/models/environment/remote_environment.rs (hunk 3 above)
Comment (mkenigs) [reply cli/flox-rust-sdk/src/models/environment/remote_environment.rs:86]: nonblocking
```suggestion
            .map_err(RemoteEnvironmentError::UpdateUpstream)?;
            // TODO: clean up git branch for temporary environment
```

File: cli/flox-rust-sdk/src/models/environment/remote_environment.rs (hunk 2 above)
Comment (mkenigs) [reply cli/flox-rust-sdk/src/models/environment/remote_environment.rs:87]: Can you also verify in a test?

Comment (ysndr) [reply cli/flox-rust-sdk/src/models/environment/remote_environment.rs:88]: it creates a temporary link `<system>.<name>.encode(tempdir)`

Comment (ysndr) [reply cli/flox-rust-sdk/src/models/environment/remote_environment.rs:88]: i mean yeah?
I guess the question is whether `--remote` will always affect the same out links in all processes or be a bit more isolated.. I see reason for both

Comment (mkenigs) [reply cli/flox-rust-sdk/src/models/environment/remote_environment.rs:88]: If you don't want to block, that's fine by me. I would probably link to `<system>.<name>` and add a test for it though. What are some downsides of using the same out link?

Comment (ysndr) [reply cli/flox-rust-sdk/src/models/environment/remote_environment.rs:109]: changed it to use the same and added a test

still blovked on #621 and #620

**Comment by zmitchell** [comment]
I gave this a once over and it looks good to me, but there are merge conflicts

**Review by mkenigs — APPROVED** [review]

**Review by mkenigs — APPROVED** [review]

**Pull request merged**